### PR TITLE
feat: introduce size 0 tag component

### PR DESCRIPTION
**Pull request by cjbell**
* Size 0 tag
* Sets different left padding
* Cannot have `onCopy` or `onRemove`

### CODE REVIEW

File: packages/icon/src/Icon/Icon.tsx [hunk 1]
```diff
@@ -64,7 +64,12 @@ const Icon = <T extends TgphElement>({
       {...props}
     >
       {IconComponent && (
-        <IconComponent aria-label={alt} width="100%" height="100%" />
+        <IconComponent
+          aria-label={alt}
+          width="100%"
+          height="100%"
+          display="block"
```
Comment (cjbell) [inline packages/icon/src/Icon/Icon.tsx:71]: When the parent container for an icon is inline-block you always need this so the icon acts appropriately in the container

File: packages/tag/src/Tag/Tag.tsx [hunk 2]
```diff
@@ -227,10 +228,12 @@ const Default = <T extends TgphElement>({
...
       {icon && <Icon {...icon} />}
       <Text as="span">{children}</Text>
-      {onRemove && (
+      {onRemove && size !== "0" && (
         <Button onClick={onRemove} icon={{ icon: Lucide.X, alt: "Remove" }} />
       )}
-      {onCopy && <CopyButton onClick={onCopy} textToCopy={textToCopy} />}
+      {onCopy && size !== "0" && (
+        <CopyButton onClick={onCopy} textToCopy={textToCopy} />
+      )}
```
Comment (kylemcd) [inline packages/tag/src/Tag/Tag.tsx:236]: I think gating this by `size` is an unnecessary abstraction for this.

File: packages/tag/src/Tag/Tag.tsx [hunk 3]
```diff
@@ -76,6 +76,7 @@ const Text = <T extends TgphElement>({
       as={as}
       size={context.size}
       color={COLOR.Text[context.variant][context.color]}
+      weight="medium"
```
Comment (cjbell) [inline packages/tag/src/Tag/Tag.tsx:79]: Tags were default to regular, which doesn't match the design

File: packages/tag/src/Tag/Tag.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ const Root = <T extends TgphElement>({
         display="inline-flex"
         align="center"
         rounded="3"
-        pl="2"
+        pl={size === "0" ? "1" : "2"}
```
Comment (kylemcd) [inline packages/tag/src/Tag/Tag.tsx:53]: Can we add this to a map in Tags.constants instead? Ideally all of these are configuration based and not inline prop logic. 

```
const rootSpacing = {
"0": "1",
"1": "2",
....
"
```

File: packages/tag/src/Tag/Tag.tsx (hunk 2 above)
Comment (cjbell) [reply packages/tag/src/Tag/Tag.tsx:236]: If we don't do that then we need to build a size 0 button fwiw

Comment (kylemcd) [reply packages/tag/src/Tag/Tag.tsx:236]: Yeah, I get that it won't work. But nothing is stopping you from doing this either. 
```
<Tag.Root size="0">
   <Tag.Button>..</Tag.Button>
</Tag.Root>
```

Just don't want to muddy up components with inline logic like this.

Comment (cjbell) [reply packages/tag/src/Tag/Tag.tsx:236]: Ah got it. So how should we resolve though? Introduce a size 0 button?

Comment (kylemcd) [reply packages/tag/src/Tag/Tag.tsx]: Yeah I think that makes the most sense, I'm sure we will run across a need for one at some point too.

Comment (kylemcd) [reply packages/tag/src/Tag/Tag.tsx]: We can add that as a TODO item in Linear and merge this for now though

**Review by kylemcd — APPROVED** [review]

**Comment by kylemcd** [comment]
Looks good, don't forget to add a changeset 👍

**Pull request merged**